core/src/main/scala/org/bitcoins/core/number/BasicArithmetic.scala

```diff
@@ -0,0 +1,36 @@
+package org.bitcoins.core.number
+
+import scala.util.Try
+
+trait BasicArithmetic[N] {
+
+  def +(n: N): N
+
+  /**
+    * Some classes have restrictions on upper bounds
+    * for it's underlying value. This might cause the `+`
+    * operator to throw. This method wraps it in a `Try`
+    * block.
+    */
+  def safeAdd(n: N): Try[N] = Try { this + n }
```

Comment:
I think this is important, we should probably do `addSafe` rather than `safeAdd`

If i am using autocomplete on my text editor I will type `a` for `add` and not see the `safeAdd` method. I think it is a little thing that is important when designing a library.

Reply:
Same thing with others